pytential/symbolic/pde/scalar.py

```diff
@@ -48,110 +66,160 @@ def __init__(self, kernel, use_l2_weighting):
+        :returns: a representation for the unknowns based on an integral
+            equation with density *u*. If *qbx_forced_limit* denotes an
+            on-surface evaluation, the corresponding jump relations are not
+            added to the representation.
+        """
+        raise NotImplementedError
+
+    def operator(self, u, **kwargs):
+        """
+        :returns: a boundary integral operator with corresponding jump
+            relations that can be used to solve for the density *u*.
+        """
+        raise NotImplementedError
+
 # }}}
 
 
 # {{{ dirichlet
 
 class DirichletOperator(L2WeightedPDEOperator):
     """IE operator and field representation for solving Dirichlet boundary
-    value problems with scalar kernels (e.g.
-    :class:`sumpy.kernel.LaplaceKernel`,
-    :class:`sumpy.kernel.HelmholtzKernel`, :class:`sumpy.kernel.YukawaKernel`)
+    value problems with scalar kernels (e.g. :class:`~sumpy.kernel.LaplaceKernel`,
+    :class:`~sumpy.kernel.HelmholtzKernel`, :class:`~sumpy.kernel.YukawaKernel`)
 
     .. note ::
 
         When testing this as a potential matcher, note that it can only
         access potentials that come from charge distributions having *no* net
-        charge. (This is true at least in 2D.)
+        charge. (This is true at least in 2D)
+
+    Inherits from :class:`L2WeightedPDEOperator`.
 
     .. automethod:: is_unique_only_up_to_constant
     .. automethod:: representation
     .. automethod:: operator
+    .. automethod:: __init__
     """
 
-    def __init__(self, kernel, loc_sign, alpha=None, use_l2_weighting=False,
-            kernel_arguments=None):
+    def __init__(self, kernel: Kernel, loc_sign: int, *,
+            alpha: Optional[Number] = None,
+            use_l2_weighting: bool = False,
+            kernel_arguments: Optional[Dict[str, Any]] = None):
         """
-        :arg loc_sign: +1 for exterior, -1 for interior
-        :arg alpha: the coefficient for the combined-field representation
-            Set to 0 for Laplace.
+        :param loc_sign: :math:`+1` for exterior or :math:`-1` for interior
+            problems.
+        :param alpha: a complex coefficient with positive imaginary part for
+            the combined-field integral representation (CFIE) for the
+            Helmholtz equation (based on Brakhage and Werner).
+            For other kernels, it does does not offer any benefits.
         """
         assert loc_sign in [-1, 1]
-
-        from sumpy.kernel import Kernel, LaplaceKernel
         assert isinstance(kernel, Kernel)
 
         if kernel_arguments is None:
             kernel_arguments = {}
-        self.kernel_arguments = kernel_arguments
-
-        self.loc_sign = loc_sign
-
-        L2WeightedPDEOperator.__init__(self, kernel, use_l2_weighting)
 
         if alpha is None:
-            if isinstance(self.kernel, LaplaceKernel):
-                alpha = 0
-            else:
+            # See Culton and Kress Chapter 3 for an explanation
```

Comment:
```suggestion
            # Use a combined-field/Brakhage-Werner representation
            # (alpha != 0) to avoid spurious resonances (mainly for
            # the exterior problem)
            # See:
            # - Brakhage and Werner.
            #    Über das Dirichletsche Außenraumproblem für die
            #    Helmholtzsche Schwingungsgleichung.
            #    https://doi.org/10.1007/BF01220037
            # - Coulton and Kress, Chapter 3
```